src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -620,23 +626,29 @@ private function addService($id, Definition $definition)
             $lazyInitialization = '';
         }
 
-        // with proxies, for 5.3.3 compatibility, the getter must be public to be accessible to the initializer
-        $isProxyCandidate = $this->getProxyDumper()->isProxyCandidate($definition);
-        $visibility = $isProxyCandidate ? 'public' : 'protected';
+        $asFile = $this->asFiles && $definition->isShared();
```

Comment:
shared services are not split because they can be created several times (that's their purpose) and that would mean paying for "require" several times (for shared services, the code needs to be loaded anyway, so there is no extra "tax")